src/Plutarch/Extra/Time.hs

```diff
@@ -0,0 +1,159 @@
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Plutarch.Extra.Time (
+    PCurrentTime (..),
+    pcurrentTime,
+    currentTime,
+    passertCurrentTime,
+    pisWithinCurrentTime,
+    pisCurrentTimeWithin,
+) where
+
+import Control.Composition ((.*))
+import GHC.Records (HasField)
+import Plutarch.Api.V1 (
+    PExtended (PFinite),
+    PInterval (PInterval),
+    PLowerBound (PLowerBound),
+    PPOSIXTime,
+    PUpperBound (PUpperBound),
+ )
+import Plutarch.Api.V2 (PPOSIXTimeRange)
+import Plutarch.Extra.Applicative (pliftA2)
+import Plutarch.Extra.Field (pletAll, pletAllC)
+import Plutarch.Extra.Maybe (passertPJust, pjust, pnothing)
+import Plutarch.Extra.TermCont (pmatchC)
+
+{- | Represent a fully bounded time range.
+
+     Note: 'PCurrentTime' doesn't need a Haskell-level equivalent because it
```

Comment:
Need more explanation the difference between `PCurrentTime` and `PPOSIXTimeRange` because they look identical and have same purpose. More emphasis DataPlut vs ScottPlut would make it more clear.

Reply:
Note that `PPOSIXTimeRange` doesn't require a time range to be bounded and closed on both sides. And also as you pointed it out, `PCurrentTime` is scott-encoded.

Reply:
Yeah. I'd be nice to mention that in docstring.